app/controllers/api/flavors_controller.rb

```diff
@@ -1,4 +1,30 @@
 module Api
   class FlavorsController < BaseController
+    def create_resource(_type, _id, data)
+      attrs = validate_flavor_attrs(data)
+      task_id = Flavor.create_flavor_queue(User.current_user.id, EmsCloud.find_by_id(attrs['ems']['id']), attrs.deep_symbolize_keys)
+      action_result(true, 'Creating Flavor', :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    def delete_resource(type, id, _data = {})
+      flavor = resource_search(id, type, collection_class(:flavors))
+      raise "Delete not supported for #{flavor_ident(flavor)}" unless flavor.respond_to?(:delete_flavor_queue)
```

Comment:
@alexander-demichev I may be incorrect, but is this method `delete_flavor_queue` the correct method? I don't see any flavor descendants responding to it